src/main/java/reposense/model/RepoLocation.java

```diff
@@ -51,31 +59,104 @@ public String getOrganization() {
         return organization;
     }
 
+    public Optional<String> getParsedBranch() {
+        return parsedBranch;
+    }
+
     /**
-     * Verifies {@code location} can be presented as a {@code URL} or {@code Path}.
-     * @throws InvalidLocationException if otherwise.
+     * Parses a String representing a repo location (which could be a repo URL, branch URL
+     * or a filepath), and returns an array containing the following details
+     * of the repository (in order):
+     * { location, repository name, organisation name, branch name (if any) }
+     *
+     * @param location a repository location, which is a file path or URL with the branch
+     *         name optionally appended
+     * @throws InvalidLocationException if the repo location is an invalid path or an invalid URL
      */
-    private void verifyLocation(String location) throws InvalidLocationException {
-        boolean isValidPathLocation = false;
-        boolean isValidGitUrl = false;
-
-        try {
-            Path pathLocation = Paths.get(location);
-            isValidPathLocation = Files.exists(pathLocation);
-        } catch (InvalidPathException ipe) {
-            // Ignore exception
+    private static String[] parse(String location) throws InvalidLocationException {
+        String[] parsedInfo = tryParsingAsRepoUrl(location);
+        if (parsedInfo != null) {
+            return parsedInfo;
         }
 
-        try {
-            new URL(location);
-            isValidGitUrl = location.endsWith(GIT_LINK_SUFFIX);
-        } catch (MalformedURLException mue) {
-            // Ignore exception
+        parsedInfo = tryParsingAsPath(location, true);
+        if (parsedInfo != null) {
+            return parsedInfo;
+        }
+
+        parsedInfo = tryParsingAsBranchUrl(location);
+        if (parsedInfo != null) {
+            return parsedInfo;
+        }
+
+        throw new InvalidLocationException(MESSAGE_INVALID_LOCATION);
+    }
+
+    /**
+     * Parses a given path to a repo and returns an array containing the following info:
+     * { url, repository name, organisation name, branch name (if any) }
+     *
+     * @param path A path to a repository
+     * @param isPathValidationNeeded If this flag is set to true, then the method will check
+     *         verify that {@code path} is a valid path and refers to an actual directory on the
+     *         filesystem.
+     *
+     * @return null if the given String is an invalid path, or no directory exists at the path,
+     *         and {@code isPathValidationNeeded} was set to true.
+     */
+    private static String[] tryParsingAsPath(String path, boolean isPathValidationNeeded)  {
```

Comment:
For the `tryParsingAsPath` method, I have modified it to accept a boolean flag `isPathValidationNeeded`.
This allows us to test `tryParsingAsPath` with mock filepaths that do not exist on the file system 
(since we only want to test whether the mock filepaths are parsed correctly)